### PR TITLE
JoinScript: add DiscoveryService installation method

### PR DESCRIPTION
**Pull request by marcoandredinis**
DiscoveryService can now be enabled using the JoinScript.

To generate a script, first create a token with `discovery` role.
You'll need a DiscoveryGroup (can be any string).
Then, build the URL as usual (it has a query parameter to indicate the DiscoveryGroup).

It should look like this:
```
https://127.0.0.1.nip.io:3080/scripts/592f8868de9d7dcb7538856552708a4e/install-discovery.sh?discoveryGroup=olamundo
```

Demo:
Generated `teleport.yaml`
image

Inventory after the instance joins the cluster:
image

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]

File: lib/web/scripts/node-join/install.sh [hunk 1]
```diff
@@ -480,6 +481,39 @@ db_service:
...
+    severity: INFO
+auth_service:
+  enabled: no
+ssh_service:
+  enabled: no
+proxy_service:
+  enabled: no
+discovery_service:
+  enabled: "yes"
+  discovery_group: "{{.discoveryGroup}}"
```
Comment (greedy52) [inline lib/web/scripts/node-join/install.sh:513]: without any matchers (e.g. `aws`, `azure`), this discovery service doesn't do anything. Is this intended?

Comment (marcoandredinis) [reply lib/web/scripts/node-join/install.sh:513]: It should be capable of watch DiscoveryConfig resource change events and, if the `discovery_group` matches, dynamically load them.

Comment (greedy52) [reply lib/web/scripts/node-join/install.sh:513]: 👍 Nice, I didn't know it can do that.

**Review by greedy52 — APPROVED** [review]

File: lib/web/scripts/node-join/install.sh (hunk 1 above)
Comment (marcoandredinis) [reply lib/web/scripts/node-join/install.sh:513]: It's quite recent
Related issue:
https://github.com/gravitational/teleport/issues/25494

**Pull request merged**